### PR TITLE
Cybereason app v2.1.0

### PR DESCRIPTION
**Pull request by gusdadiya-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
- [x] Limit your pull request to one application and submit multiple pull requests if necessary.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)

- New action to restart a sensor using the pylumID provided as an input
- New action to upgrade a sensor using the pylumID provided as an input
- New action to isolate/un-isolate machines that are not part of a specific malop
- New action to retrieve data about machines based on the machine name/IP

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- We have closed this phantomcyber MR[(!737)](https://github.com/phantomcyber/phantom-apps/pull/737) and raised against splunk-soar-connector

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: cybereason_poller.py [hunk 1]
```diff
@@ -32,35 +40,49 @@ def do_poll(self, connector, param):
...
             self.cr_session = CybereasonSession(connector).get_session()
             malop_start_time_microsec_timestamp = round(malop_start_time.timestamp() * 1000)
             # When called as a scheduled poll, max_container count comes as 4294967295 which causes a Cybereason API error.
             container_count = min(int(param.get(phantom.APP_JSON_CONTAINER_COUNT)), 5000)
-            success = success & self._fetch_and_ingest_malops(connector, config, malop_start_time_microsec_timestamp, container_count)
-            success = success & self._fetch_and_ingest_malwares(connector, config, malware_millisec_since_last_poll, container_count)
+            success = success & self._fetch_and_ingest_malops(connector, config,
+                    malop_start_time_microsec_timestamp, container_count)
+            success = success & self._fetch_and_ingest_malwares(connector, config,
+                    malware_millisec_since_last_poll, container_count)
```
Comment (paulnawat) [inline cybereason_poller.py]: nit: If it's not exceeded the line length, Should we move all params to the next line? 
```python
 success = success & self._fetch_and_ingest_malops(
                  connector, config, malop_start_time_microsec_timestamp, container_count)
```

File: cybereason.json [hunk 2]
```diff
@@ -1576,7 +1817,7 @@
                 {
                     "data_path": "action_result.data.*.dns_query",
                     "data_type": "string",
-                    "column_name": "DNS Query",
+                    "column_name": "Dns Query",
```
Comment (paulnawat) [inline cybereason.json]: nit: I think the capital `DNS` is the correct version, right?

File: cybereason_connector.py [hunk 3]
```diff
@@ -600,6 +705,251 @@ def _get_malop_sensor_ids(self, malop_id, action_result):
...
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            self.save_progress(err)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Error occurred. {}".format(err)), [])
+
+        return RetVal(action_result.set_status(phantom.APP_SUCCESS), sensor_ids)
+
+    def _handle_upgrade_sensor(self, param):
+        self.save_progress("In _handle_upgrade_sensor function")
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
```
Comment (paulnawat) [inline cybereason_connector.py]: nit: Should we just keep one of these?

File: cybereason_connector.py [hunk 4]
```diff
@@ -600,6 +705,251 @@ def _get_malop_sensor_ids(self, malop_id, action_result):
...
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully requested for sensor restart")
+
+    def _get_machine_name_by_machine_ip(self, machine_ip, action_result):
+        machine_names = []
+        try:
+            cr_session = CybereasonSession(self).get_session()
+
+            url = "{0}/rest/sensors/query".format(self._base_url)
+            self.save_progress(url)
```
Comment (paulnawat) [inline cybereason_connector.py]: suggestion: Should be add some additional text here to show what the url is about?

File: cybereason_connector.py [hunk 5]
```diff
@@ -600,6 +705,251 @@ def _get_malop_sensor_ids(self, malop_id, action_result):
...
+
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, "Error occurred. {}".format(err))
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully requested for sensor upgrade")
+
+    def _handle_restart_sensor(self, param):
+        self.save_progress("In _handle_restart_sensor function")
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
```
Comment (paulnawat) [inline cybereason_connector.py]: Should we keep just one of theses?

**Comment by paulnawat** [comment]
@gusdadiya-crest  is this app migrated from Gitlab? Could you add a few description for this PR?

**Review by paulnawat — DISMISSED** [review]
lgtm, I left some minor comments / suggestions.

File: cybereason_poller.py (hunk 1 above)
Comment (gusdadiya-crest) [reply cybereason_poller.py]: Updated

File: cybereason.json (hunk 2 above)
Comment (gusdadiya-crest) [reply cybereason.json]: Fixed

File: cybereason_connector.py (hunk 3 above)
Comment (gusdadiya-crest) [reply cybereason_connector.py]: Removed

File: cybereason_connector.py (hunk 4 above)
Comment (gusdadiya-crest) [reply cybereason_connector.py]: Added

File: cybereason_connector.py (hunk 5 above)
Comment (gusdadiya-crest) [reply cybereason_connector.py]: Removed

**Review by paulnawat — APPROVED** [review]
lgtm. Thank you for making the updates!

**Pull request merged**